src/Engine/Manager/RenderManager.cpp

```diff
@@ -243,82 +251,88 @@ void RenderManager::Render(World& world, const glm::mat4& translationMatrix, con
-                Entity* entity = mesh->entity;
-                Material* material = entity->GetComponent<Material>();
-                Controller* controller = entity->GetComponent<Controller>();
-                if (material != nullptr) {
-                    if (controller != nullptr && hmdRenderSurface != nullptr) {
-                        glm::mat4 ctrlModelMatrix = controller->HandleTransformation(entity);
-                        renderer->RenderStaticMesh(mesh->geometry, material->albedo->GetTexture(), material->normal->GetTexture(), material->metallic->GetTexture(), material->roughness->GetTexture(), ctrlModelMatrix, false);
+    {
+        PROFILE("Render static meshes");
+        {
+            GPUPROFILE("Render static meshes", Video::Query::Type::TIME_ELAPSED);
+            {
+                GPUPROFILE("Render static meshes", Video::Query::Type::SAMPLES_PASSED);
+
+                // Cull lights and update light list.
+                LightWorld(world, viewMatrix, projectionMatrix, viewProjectionMatrix);
+
+                // Push matricies and light buffer to the GPU.
+                renderer->PrepareStaticMeshRendering(viewMatrix, projectionMatrix);
+
+                // Render meshes.
+                for (Mesh* mesh : meshComponents) {
+                    if (mesh->IsKilled() || !mesh->entity->enabled)
+                        continue;
+
+                    if (mesh->geometry != nullptr && mesh->geometry->GetType() == Video::Geometry::Geometry3D::STATIC) {
+                        Entity* entity = mesh->entity;
+                        Material* material = entity->GetComponent<Material>();
+                        Controller* controller = entity->GetComponent<Controller>();
+                        if (material != nullptr) {
+                            if (controller != nullptr && hmdRenderSurface != nullptr) {
+                                glm::mat4 ctrlModelMatrix = controller->HandleTransformation(entity);
+                                renderer->RenderStaticMesh(mesh->geometry, material->albedo->GetTexture(), material->normal->GetTexture(), material->metallic->GetTexture(), material->roughness->GetTexture(), ctrlModelMatrix, false);
+                            } else
+                                renderer->RenderStaticMesh(mesh->geometry, material->albedo->GetTexture(), material->normal->GetTexture(), material->metallic->GetTexture(), material->roughness->GetTexture(), entity->GetModelMatrix(), false);
+                        }
                     }
-                    else
-                        renderer->RenderStaticMesh(mesh->geometry, material->albedo->GetTexture(), material->normal->GetTexture(), material->metallic->GetTexture(), material->roughness->GetTexture(), entity->GetModelMatrix(), false);
                 }
             }
         }
     }
-    }
-    }
     renderSurface->GetShadingFrameBuffer()->Unbind();
 
     /// @todo Render skinned meshes.
-    
+
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The function 'GetDirectionalLights' is never used. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://localhost:9000/coding_rules#rule_key=cppcheck%3AunusedFunction)